extensions/amp-document-recommendations/0.1/amp-document-recommendations.js

```diff
@@ -229,6 +227,42 @@ export class AmpDocumentRecommendations extends AMP.BaseElement {
     this.appendNextArticle_();
     return Promise.resolve();
   }
+
+  /**
+   * Called from a {@link AmpDocumentRecommendationsMarker} element's
+   * layoutCallback, indicating the next article should be prerendered.
+   */
+  markerLayoutCallback() {
+    if (this.nextArticle_ < this.config_.recommendations.length) {
+      this.appendNextArticle_();
+    }
+  }
 }
 
 AMP.registerElement(TAG, AmpDocumentRecommendations, CSS);
+
+/** @const */
+const MARKER_TAG = 'amp-document-recommendations-marker';
+
+class AmpDocumentRecommendationsMarker extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(unused) {
+    return true;
+  }
+
+  /** @override */
+  buildCallback() {
+    // TODO(peterjosling): reuse this as the fallback button
+    this.element.textContent = 'Loading...';
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.element.textContent = '';
+    activeInstance_.markerLayoutCallback();
```

Comment:
I don't think these should be connected through a singleton reference unless it's enforced somehow that there will be only one `amp-document-recommendations` on the page. For example, if someone wanted two articles in parallel on the page. I would consider a generated `class` or `id` reference instead. What do you think?